packages/styles/src/shellbar.scss

```diff
@@ -135,163 +196,283 @@ $block: #{$fd-namespace}-shellbar;
-      outline: none;
-    }
+        &:placeholder-shown {
+          & ~ .#{$block}__search-cancel {
+            display: none;
+          }
+        }
 
-    &-input {
-      border: none;
-      color: var(--fdShellbar_Input_Color);
-      position: relative;
+        &:not(:placeholder-shown) {
+          & ~ .#{$block}__search-submit {
+            .fd-button {
+              background-color: var(--fdShellbar_Search_Active_Button_Background);
+              color: var(--fdShellbar_Search_Active_Button_Color);
+
+              &:hover {
+                background-color: var(--fdShellbar_Search_Active_Button_Hover_Background);
+              }
+            }
+          }
+        }
 
-      @include fd-hover() {
-        &,
-        & + .#{$block}__input-group-addon {
-          background: transparent;
-          background-color: var(--fdShellbar_Input_Hover_Background) !important;
+        &--select {
+          @include fd-flex-vertical-center();
+
+          font-style: normal;
+          height: 100%;
+          flex: 0 1;
         }
       }
 
-      &::placeholder {
-        color: var(--fdShellbar_Input_Placeholder_Color);
-        font-style: italic;
-      }
+      &-addon {
+        @include fd-flex-vertical-center() {
+          height: 100%;
+          flex: 0 1;
+        }
 
-      &--select {
-        @include fd-flex-vertical-center();
+        .#{$block}__button {
+          height: var(--fdShellbar_Input_Addon_Dimension);
+          min-width: var(--fdShellbar_Input_Addon_Dimension);
+          border-radius: var(--fdShellbar_Input_Border_Radius);
+          border-width: var(--fdShellbar_Input_Addon_Border_Width);
+        }
 
-        font-style: normal;
-        height: 100%;
+        + & {
+          background: red;
+        }
       }
-    }
 
-    &-addon {
-      @include fd-flex-vertical-center() {
-        height: 100%;
-      }
+      &-category {
+        flex-shrink: 0;
+
+        @include fd-hover() {
+          background-color: var(--fdShellbar_Input_Background) !important;
+          & ~ .#{$block}__search-field-input,
+          & ~ .#{$block}__search-field-addon {
+            --fdInputGroup_Addon_Background: var(--fdShellbar_Input_Background);
+            --fdInputGroup_Input_Background: var(--fdShellbar_Input_Background);
+          }
 
-      .#{$block}__button {
-        height: var(--fdShellbar_Input_Addon_Dimension);
-        min-width: var(--fdShellbar_Input_Addon_Dimension);
-        border-radius: var(--fdShellbar_Input_Border_Radius);
-        border-width: var(--fdShellbar_Input_Addon_Border_Width);
+          .#{$block}__search-category {
+            box-shadow: var(--sapContent_Interaction_Shadow);
+          }
+
+          .fd-select__button {
+            background-color: var(--fdShellbar_Search_Category_Button_Hover_Background);
+          }
+        }
+
+        .fd-select {
+          @include fd-set-margin-right(var(--fdShellbar_Search_Category_Offset_Right));
+        }
```

Comment:
These 2 rules you can't use as it violates the self-contained rules. You are using as a selector a class that belongs to another component